From ba365841c980d77f8ad180bd8a2cd3cd3b5193ee Mon Sep 17 00:00:00 2001
From: Raja Ganapathi Busam <quic_rbusam@quicinc.com>
Date: Wed, 16 Apr 2025 17:55:42 +0530
Subject: [PATCH] v4l2: consider Q08C as contiguous planes

Issue:
By default, the first match of gst_v4l2_formats is considered
as non-contiguous planes. And the next match (fallback) as contiguous.
But Q08C is contiguous planes and doesn't have non-contiguous equivalant.

Fix:
Consider the Q08C as contiguous planes. Re-assign the fourcc values.

Upstream-Status: Inappropriate [Qualcomm specific]

Signed-off-by: Raja Ganapathi Busam <quic_rbusam@quicinc.com>
---
 sys/v4l2/gstv4l2object.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index d7fcffb..f14df75 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -1873,6 +1873,11 @@ gst_v4l2_object_get_caps_info (GstV4l2Object * v4l2object, GstCaps * caps,
       fourcc_nc = desc->v4l2_format;
     if (fallback_desc)
       fourcc = fallback_desc->v4l2_format;
+
+    if (fourcc_nc == V4L2_PIX_FMT_QC08C) {
+      fourcc_nc = 0;
+      fourcc = V4L2_PIX_FMT_QC08C;
+    }
   } else if (g_str_equal (mimetype, "video/mpegts")) {
     fourcc = V4L2_PIX_FMT_MPEG;
   } else if (g_str_equal (mimetype, "video/x-dv")) {
-- 
2.34.1

